typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -128,31 +137,89 @@ export abstract class HyperlaneDeployer<
 
     this.logger(`Deploy ${contractName} on ${chain}`);
 
-    const contract = await factory
-      .connect(signer)
-      .deploy(...args, chainConnection.overrides);
+    if (
+      deployOpts &&
+      deployOpts.create2Salt &&
+      (await chainConnection.provider.getCode(CREATE2FACTORY_ADDRESS)) != '0x'
+    ) {
+      if (args.length > 0) {
+        throw new Error("Can't use CREATE2 with deployment args");
+      }
+      this.logger(`Deploying with CREATE2 factory`);
+
+      const create2Factory = Create2Factory__factory.connect(
+        CREATE2FACTORY_ADDRESS,
+        signer,
+      );
+      const salt = ethers.utils.keccak256(
+        ethers.utils.hexlify(ethers.utils.toUtf8Bytes(deployOpts.create2Salt)),
+      );
+      const contractAddr = await create2Factory.deployedAddress(
+        factory.bytecode,
+        await signer.getAddress(),
+        salt,
+      );
 
-    await chainConnection.handleTx(contract.deployTransaction);
+      const deployTx = deployOpts.initCalldata
+        ? await create2Factory.deployAndInit(
+            factory.bytecode,
+            salt,
+            deployOpts.initCalldata,
+          )
+        : await create2Factory.deploy(
+            factory.bytecode,
+            salt,
+            chainConnection.overrides,
+          );
+      await chainConnection.handleTx(deployTx);
 
-    const verificationInput = getContractVerificationInput(
-      contractName,
-      contract,
-      factory.bytecode,
-    );
-    this.verificationInputs[chain].push(verificationInput);
-    return contract as ReturnType<F['deploy']>;
+      this.verificationInputs[chain].push({
+        name: contractName,
+        address: contractAddr,
+        isProxy: false,
+        constructorArguments: '',
+      });
+
+      return factory.attach(contractAddr) as ReturnType<F['deploy']>;
+    } else {
+      const contract = await factory
+        .connect(signer)
+        .deploy(...args, chainConnection.overrides);
+
+      await chainConnection.handleTx(contract.deployTransaction);
+
+      if (deployOpts?.initCalldata) {
```

Comment:
oh I guess it doesn't throw because you wanted to support initialization beyond the Create2 factory?